torch/fx/symbolic_trace.py

```diff
@@ -188,6 +188,38 @@ def forward(*args, **kwargs):
             torch.nn.Module.__call__ = orig_call
         return self.graph
 
+def wrap(orig_function):
+    """
+    A decorator that can be put on existing functions to use them in fx.
+
+        @torch.fx.wrap
+        def my_custom_function(x, y):
+            return x * x + y * y
+
+        sqrt = torch.fx.wrap(math.sqrt)
+
+    When called during symbolic tracing, the function will be inserted in the
+    graph rather than tracing it. When called outside of tracing,
+    the function will execute normally.
+    """
+
+    def wrapped(*args, **kwargs):
+        proxy = None
+
+        def find_proxy(x):
+            nonlocal proxy
+            if isinstance(x, Proxy):
```

Comment:
In some cases x is a more complicated datastructure (say a dictionary) which includes a proxy but is not a proxy. In such cases this fails to find the proxy.
I suggest an alternative way to find the tracer object. Add an active_tracer class member to the Tracer object:
```py
class Tracer(TracerBase):

    active_tracer = None

    def trace(self, root: Union[torch.nn.Module, Callable]) -> Graph:
        ....
	try:
            Tracer.active_tracer = self
            ...
            torch.nn.Module.__call__ = module_call_wrapper
            ....

        finally:
            Tracer.active_tracer = None
            ...

```

Reply:
`find_proxy` is used as a callback to `map_aggregate`. `map_aggregate` does the traversal over complex data structures internally and just calls this callback for each non-aggregate value during the traversal. So this should correctly find and proxy in an arbitrarily nested aggregate